### DISCUSSION TITLE
Improve pgprtdbg output

### DISCUSSION THREAD

**Discussion by ocabra** [jesperpedersen/pgprtdbg#4] [Ideas]
The discussion about this idea started here https://github.com/jesperpedersen/pgprtdbg/discussions/2.

I believe this discussion should be about all ideas that could improve the output of the program, that could improve the analysis of the exchanged packets.

As suggested by @jesperpedersen, "one idea is have a `.bin` output where each packet is recorded in ASCII - one per line - , so like `<pid>-fe.bin`:"
```
51000000114449534341524420414C4C3B00
510000002253454C454354202A2046524F4D20706762656E63685F74656C6C65727300
```

We could output to a structured `.csv` file with the details about each message.

**Comment by jesperpedersen** [comment]
> We could output to a structured .csv file with the details about each message.

Well, that is sort of what the `output` configuration parameter and `protocol.c` deals with -- this still need a lot of work...

My idea was just to have each protocol message on a separate line in hex format

**Reply by ocabra** [reply]
Is the idea to store the messages on a buffer and output them segmented by client, e.g.:

```
Client: 1
Messages:
51000000114449534341524420414C4C3B00
510000002253454C454354202A2046524F4D20706762656E63685F74656C6C65727300
...
----------------------------------
Client: 2
...
```

or output them as they come, e.g.

```
51000000114449534341524420414C4C3B00
510000002253454C454354202A2046524F4D20706762656E63685F74656C6C65727300
...
```

**Reply by jesperpedersen** [reply]
Lets see how per client looks like
